server/datastore/mysql/software_test.go

```diff
@@ -516,79 +600,156 @@ func testSoftwareList(t *testing.T, ds *Datastore) {
+		opts.MatchQuery = "0.0.3"
+		software = listSoftwareCheckCount(t, ds, 2, 2, opts, true)
 		expected = []fleet.Software{foo003, bar003}
 		test.ElementsMatchSkipID(t, software, expected)
+
 		// query by version (case insensitive)
-		software = listSoftwareCheckCount(t, ds, 1, 1, fleet.SoftwareListOptions{ListOptions: fleet.ListOptions{MatchQuery: "V0.0.2"}}, true)
+		opts.MatchQuery = "V0.0.2"
+		software = listSoftwareCheckCount(t, ds, 1, 1, opts, true)
 		expected = []fleet.Software{foo002}
 		test.ElementsMatchSkipID(t, software, expected)
 	})
 
-	t.Run("can order by name and id", func(t *testing.T) {
-		software := listSoftwareCheckCount(t, ds, 4, 4, fleet.SoftwareListOptions{ListOptions: fleet.ListOptions{OrderKey: "name,id", OrderDirection: fleet.OrderAscending}}, false)
+	t.Run("order by name and id", func(t *testing.T) {
+		opts := fleet.SoftwareListOptions{
+			ListOptions: fleet.ListOptions{
+				OrderKey:       "name,id",
+				OrderDirection: fleet.OrderAscending,
+			},
+		}
+		software := listSoftwareCheckCount(t, ds, 5, 5, opts, false)
 		assert.Equal(t, bar003.Name, software[0].Name)
 		assert.Equal(t, bar003.Version, software[0].Version)
-		assert.Equal(t, bar003.Source, software[0].Source)
 
-		// it's ordered by id, descending
-		assert.Greater(t, software[2].ID, software[1].ID)
+		assert.Equal(t, baz001.Name, software[1].Name)
+		assert.Equal(t, baz001.Version, software[1].Version)
+
+		// foo's ordered by id, descending
 		assert.Greater(t, software[3].ID, software[2].ID)
+		assert.Greater(t, software[4].ID, software[3].ID)
 	})
 
-	t.Run("hosts count", func(t *testing.T) {
+	t.Run("order by hosts_count", func(t *testing.T) {
 		defer TruncateTables(t, ds, "software_host_counts")
 		listSoftwareCheckCount(t, ds, 0, 0, fleet.SoftwareListOptions{WithHostCounts: true}, false)
 
 		// create the counts for those software and re-run
 		require.NoError(t, ds.CalculateHostsPerSoftware(context.Background(), time.Now()))
-		software := listSoftwareCheckCount(t, ds, 4, 4, fleet.SoftwareListOptions{ListOptions: fleet.ListOptions{OrderKey: "hosts_count", OrderDirection: fleet.OrderDescending}, WithHostCounts: true}, false)
+		software := listSoftwareCheckCount(t, ds, 5, 5, fleet.SoftwareListOptions{ListOptions: fleet.ListOptions{OrderKey: "hosts_count", OrderDirection: fleet.OrderDescending}, WithHostCounts: true}, false)
 		// ordered by counts descending, so foo003 is first
 		assert.Equal(t, foo003.Name, software[0].Name)
 		assert.Equal(t, 2, software[0].HostsCount)
 	})
+
+	t.Run("order by epss_probability", func(t *testing.T) {
+		opts := fleet.SoftwareListOptions{
+			ListOptions: fleet.ListOptions{
+				OrderKey:       "epss_probability",
+				OrderDirection: fleet.OrderDescending,
+			},
+			IncludeCVEScores: true,
+		}
+
+		software := listSoftwareCheckCount(t, ds, 5, 5, opts, false)
+		assert.Equal(t, foo001.Name, software[0].Name)
+		assert.Equal(t, foo001.Version, software[0].Version)
```

Comment:
Worth checking that the returned software has the CVE scores included (`IncludeCVEScores: true`), and worth adding the inverse (`IncludeCVEScores: false` and the returned to have unset fields).

Reply:
Added test in 8a43c0bb7